doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+
+### Line 2: Error code
+
+Reasons to keep it:
+* Probably could be used to categorize errors such it is done in the http protocol
+* Good to use for shellrecorder scripts to test for errors
+* Possibility for automation (eg. if a certain error number occurred you could automatically trigger certain actions)
+
+Reasons against it:
+* Extra line
+
+Suggestion:
+Giving an error a number like it is done in C programs as return code seems to be outdated for elektra, especially when
+the reason is given anyway. Personally I never read the error number because it has not much value.
+The *reasons to keep* will also be very limited when the number of errors are significantly reduced.
+I would also suggest to remove it.
+
+### Line 1: General sentence
+
+All in all this sentence can stay because it gives the user a first hint on what happened.
+I would though change the wording and include more relevant information in it.
+I would change it like this:
+"Module <plugin> issued a <error/warning/fatal> while accessing the key database with the info:"
+where plugin should have a different color (blue?) and warning/error/fatal (yellow, red, dark red) too depending on the severity.
+
+With this line 5/6 can still be present but in a much more efficient and succinct way.
+
+### Error/Warning/Fatal etc. splitting in the specification file
+
+I would suggest to remove this concept from the specification and let the developer use a respective method to indicate if it is a warning/error/etc.
+The macros ELEKTRA_SET_ERROR, ELEKTRA_ADD_WARNING ... should be enough to make developers cautious of wrong method usage.
+What I have seen is that the return code is the only matter if it is an error or not.
+
+### Additional information which is not present
+
+Some errors can also guide users to fix the error by suggesting a possible solution:
+* Retry as sudo
+* Fix file permissions
+* Check if the file exists
+* etc.
+Developers though have to be very careful when suggesting solutions because it can easily mislead users. For many cases though, 
+this can help users a lot.
+I would suggest an optional extra line (which is only printed if the developer adds a metafield `solution` to the parentkey):
+`Possible Solution: ...`
+
+# Conclusion
+
+With the new concept we would not require a `specification` file at all which helps developers immensely but also keeps the
+most relevant information for the users. An error like this:
+
+```
+1. The command kdb set failed while accessing the key database with the info:
+2. Sorry, the error (#121) occurred ;(
+3. Description: validation failed
+4. Reason: Validation of key "<key>" with string "<value>" failed.
+5. Ingroup: plugin
+6. Module: enum
+7. At: ....../src/plugins/enum/enum.c:218
+8. Mountpoint: <parentKey>
+9. Configfile: ...../<file>.25676:1549919217.284067.tmp
+```
+
+would be changed to this:
+```
+1. Plugin enum issued a error while accessing the key database with the info:
```

Comment:
Sorry, the plugin "enum" issued error `#<NR>`:

("enum" and  `#<NR>` in colors)

Reply:
see discussion above